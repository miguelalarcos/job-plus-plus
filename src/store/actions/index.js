import { getMessages } from '@/api'

export default {
    newError(context, msg) {
        context.commit('newError', msg)
        setTimeout(() => {
          context.commit('newError', '')
        }, 3000)
    },
    appendMessageAction(context, {candidature, msg}) {
        context.commit('appendMessage', {id: Math.random(), msg, candidature, owner: 'miguel'})
    },
    async getMessagesAction(context, {candidature}){
        //const messages = await getMessages(candidature)
        //context.commit('setMessages', {messages})
    }
}