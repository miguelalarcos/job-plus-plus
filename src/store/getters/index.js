export default {
    /*totalNotifications(state){
        const a1 = Object.values(state.events)
        const ret = a1.reduce((a, b) => a + b, 0)
        return ret 
    },*/
    totalCandidateNotifications(state) {
        let count = 0
        for(let key in state.events) {
            if(key.match(/NEW_CANDIDATE/)){
                count ++
            }
        }
        return count
    },
    totalMessageNotifications(state) {
        let count = 0
        for(let key in state.events) {
            if(key.match(/MESSAGE/)){
                count += state.events[key]
            }
        }
        return count
    },
    newCandidates: (state) => (txt) =>{
        txt = txt + '.NEW_CANDIDATE' 
        const keys = Object.keys(state.events)
        
        let sum = 0
        keys.map((k)=>{
            if(k.startsWith(txt)){
                sum += state.events[k]
            }
        })
        return sum;
    },
    newMessages: (state) => (prefix) =>{
        const txt = prefix + '.MESSAGE' 
        const keys = Object.keys(state.events)
        
        let sum = 0
        keys.map((k)=>{
            if(k.startsWith(txt)){
                sum += state.events[k]
            }
        })
        return sum;
    },
    candidateNewEvents: (state) => (candidature) => {
        const events = state.candidateNewEvents.filter(x => x._id === candidature)
        if(events.length == 1) 
            return events[0].total
        else 
            return 0
    },
    offererNewEvents: (state) => (offer) => {
        const events = state.offererNewEvents.filter(x => x._id === offer)
        if(events.length == 1) 
            return events[0].total
        else 
            return 0
    }
}
