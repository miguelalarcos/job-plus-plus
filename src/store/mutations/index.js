import moment from 'moment'
import { newError } from './errors'

export default {
    setNotifications(state, count){
        state.notifications = count
    },
    newError,
    appendMessage(state, {id, msg, candidature, owner}) {
        msg = {date: moment().unix(), id, candidature, owner, msg}
        state.messages = [...state.messages, msg]
    },
    setMessages(state, {messages}) {
        state.messages = [...messages]
    },
    setMark(state, {index, mark}) {
        let array = state.evaluationForm
        state.evaluationForm = array.map((item, i) => {
            if(index !== i) return item
            return {...item, mark}
        })
    },
    candidateSavesObservation(state, {id, observations}){
        const item = {...state.myCandidatures[id], observations}
        state.myCandidatures = {...state.myCandidatures, [id]: item}
    },
    candidateSetsCandidature(state, {id, payload}) {
        const item = {...state.myCandidatures[id], ...payload}
        state.myCandidatures = {...state.myCandidatures, [id]: item}
    }
}
