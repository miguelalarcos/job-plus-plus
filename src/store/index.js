import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)
import getters from '@/store/getters'
import mutations from '@/store/mutations'
import actions from '@/store/actions'

export default new Vuex.Store({
  state: {
    loading: false,
    //name: 'miguel',
    user: {
      //user_id: '5b4a116e2427e5423fcbfac4',
      //experience: []
    },
    errorMessage: "",
    successMessage: "",
    listOfTags: [],
    observationsSaved: {},
    myOffersSaved: {},
    candidateNewEvents: [],
    offererNewEvents: [], //{},
    offererNewCandidates: {},
    candidate: {},
    experiences: {},
    evaluationForm: 
      [
        {id: '0', question: "sabes programar en react?", answer: "https://jsfiddle.net/", mark: 0, observations: ''},
        {id: '1', question: "sabes programar en vue?", answer: "no", mark: 0, observations: ''}
      ],
    fillForm: [
      {question: "sabes programar en react?", answer: ""},
      {question: "sabes programar en vue?", answer: ""}
    ],
    candidatureSelected: {},
    myCandidatures: {},
    candidaturesForOfferer: {},
    myOffers: {},
    searchOffers: {
      /*'"5b3d58088b4ae963d3656009': {
        "_id" : "5b3d58088b4ae963d3656009",
        "title" : "oferta python",
        "description" : "una oferta en python basada en bottle",
        "province" : "Murcia",
        "offerer" : "miguel",
        "remote" : false,
        "status" : "open",
        "tags" : [ 
            "python", 
            "vue.js"
        ]
    }*/
    },
  },
  getters:{
    ...getters
  },
  mutations: {
    ...mutations
  },
  actions: {
    ...actions
  }
})
