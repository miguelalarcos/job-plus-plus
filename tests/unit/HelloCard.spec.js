import { mount } from '@vue/test-utils'
import MainCard from '@/components/MainCard.vue'
import store from '@/store'

describe('MainCard.vue', () => {  
  it('renders main card with 3 notifications', () => {
    const wrapper = mount(MainCard, {
      store
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders main card with 1 notification', () => {
    store.state.totalNotifications = 1
    const wrapper = mount(MainCard, {
      store
    })

    expect(wrapper.element).toMatchSnapshot()
  })

})
