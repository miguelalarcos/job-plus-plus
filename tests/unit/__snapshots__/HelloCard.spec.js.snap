// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainCard.vue renders main card with 1 notification 1`] = `
<div
  class="hello"
>
  <!---->
   
  <h2>
    Hola miguel, tienes una nueva notificaci√≥n.
  </h2>
   
  <!---->
</div>
`;

exports[`MainCard.vue renders main card with 3 notifications 1`] = `
<div
  class="hello"
>
  <h2>
    Hola miguel, tienes 3 nuevas notificaciones.
  </h2>
   
  <!---->
   
  <!---->
</div>
`;
